test/NearestNeighborsModule.jl

```diff
@@ -0,0 +1,34 @@
+@testset "nearest_neighbors" begin
+    x = [1., 2., 3.5, 4.2, 5.9, 8.]
+    kern = SqExponentialKernel()
+    fx = GP(kern)(x, 0.0);
+    x2 = 1.0:0.1:8
+    y = sin.(x)
+
+    @testset "Using all neighbors is the same as the exact GP" begin
+        opt_pred = mean_and_cov(posterior(NearestNeighbors(length(x) - 1),
+            fx, y)(x2))
+        pred = mean_and_cov(posterior(fx, y)(x2))
+        for i in 1:2
+            @test all(isapprox.(opt_pred[i], pred[i]; atol=1e-4))
+        end
+    end
+
+    @testset "Using nearest neighbors approximates the exact GP" begin
+        opt_pred = mean_and_cov(posterior(NearestNeighbors(3), fx, y)(x2))
+        pred = mean_and_cov(posterior(fx, y)(x2))
+        for i in 1:2
+            @test all(isapprox.(opt_pred[i], pred[i]; atol=1e-1))
+        end
+    end
+
+    @testset "Zygote can take gradients of the logpdf" begin
+        function objective(lengthscale, var)
```

Comment:
Does this function use the Vecchia approximation at all? It looks to me like it just does exact inference.

Reply:
You're quite correct- I entirely forgot to include the Vecchia approximation here. I should have just pushed a fix for this.